hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -104,7 +103,8 @@
     public static final String SERVICE_NAME = "hz:core:clientEngine";
 
     private static final int EXECUTOR_QUEUE_CAPACITY_PER_CORE = 100000;
-    private static final int THREADS_PER_CORE = 20;
+    private static final int BLOCKING_THREADS_PER_CORE = 20;
+    private static final int THREADS_PER_CORE = 1;
```

Comment:
Is this pr final ? reducing threads per core without moving blocking tasks is dangerous. 
Lots of messages will wait for threads to be available unnecessarily.

Reply:
Nice catch! I missed this change during patching older version,
I'll then set 5 (blocking) + 15, does it sound reasonable, @sancar?

Reply:
I think we should not change the `THREADS_PER_CORE` and cause regression. 
Setting a new one 5 will mean that, only 5 clients can do authentication at a time. Not sure the value should be. The main purpose is not to block migration threads anyway. So 5 here should be ok.